champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/Skill.java

```diff
@@ -96,16 +98,36 @@ public void reload() {
         }
     }
 
-    protected <T> T getConfig(String name, Object defaultValue, Class<T> type) {
+    private String getPath(String name) {
         String path;
         if (getClassType() != null) {
             path = "skills." + getClassType().name().toLowerCase() + "." + getName().toLowerCase().replace(" ", "") + "." + name;
         } else {
             path = "skills.global." + getName().toLowerCase().replace(" ", "") + "." + name;
         }
-        return champions.getConfig().getOrSaveObject(path, defaultValue, type);
+        return path;
+    }
+
+    protected <T> T getConfig(String name, Object defaultValue, Class<T> type) {
+        return champions.getConfig().getOrSaveObject(getPath(name), defaultValue, type);
     }
 
+    /**
+     * @param name name of the value
+     * @param defaultValue default value
+     * @param type The type of default value
+     * @param <T> The type of default value
+     * @return returns the config value if exists, or the default value if it does not. Does not save value in the config
+     */
+    protected <T> T getNonDefaultConfig(String name, Object defaultValue, Class<T> type) {
+        ExtendedYamlConfiguration config = champions.getConfig();
+        if (config.isSet(getPath(name))) {
+            return config.getObject(getPath(name), type);
+        }
+        return (type.cast(defaultValue));
+    }
```

Comment:
You could put this in ExtendedYamlConfiguration (this is our own implementation)

You could likely just use config.get(key, defaultValue) that comes with the spigot config as well

Reply:
config#getObject() works the best, changed to that